src/pages/loading-waiting-states/spinner-overlays/index.tsx

```diff
@@ -0,0 +1,209 @@
+    const [eulaAccepted, setEulaAccepted] = useState(false);
+    const [eulaLoaded, setEulaLoaded] = useState(false);
+
+    const changeCheckboxState = (event: React.ChangeEvent<HTMLInputElement>): void => {
+        setEulaAccepted(event.target.checked);
+    };
+
+    useEffect(() => {
+        let timer: NodeJS.Timeout;
+        if (!eulaLoaded) {
+            timer = setTimeout(() => {
+                setEulaLoaded(true);
+            }, 3000);
+        }
+        return (): void => {
+            clearTimeout(timer);
+        };
+    }, [eulaLoaded]);
+
+    const reloadEulaDetails = useCallback((): void => {
+        setEulaLoaded(false);
+        setEulaAccepted(false);
+    }, []);
+
+    return (
+        <div style={{ minHeight: '100vh' }}>
+            <AppBar data-cy="pxb-toolbar" position={'sticky'}>
+                <Toolbar classes={{ gutters: classes.toolbarGutters }}>
+                    <Hidden mdUp={true}>
+                        <IconButton
+                            data-cy="toolbar-menu"
+                            color={'inherit'}
+                            onClick={(): void => {
+                                dispatch({ type: TOGGLE_DRAWER, payload: true });
+                            }}
+                            edge={'start'}
+                            style={{ marginRight: 20 }}
+                        >
+                            <MenuIcon />
+                        </IconButton>
+                    </Hidden>
+                    <Typography variant={'h6'} color={'inherit'}>
+                        Spinner Overlays
+                    </Typography>
+                    <div />
+                </Toolbar>
+            </AppBar>
+            <div className={classes.container}>
+                <Card className={classes.card}>
+                    {!eulaLoaded && (
+                        <div className={classes.overlayBackground}>
+                            <CircularProgress size={96} variant={'indeterminate'} color="secondary" />
+                        </div>
+                    )}
+                    <CardHeader
+                        title={
+                            <Typography variant={'h6'} style={{ fontWeight: 600 }}>
+                                {eulaDetails.title}
+                            </Typography>
+                        }
+                        className={classes.cardTitle}
+                    />
+                    <CardContent className={classes.cardContent}>
+                        {eulaLoaded ? (
+                            <Typography className={classes.eulaContent}>
+                                {eulaLoaded && eulaDetails.eulaContent}
+                            </Typography>
+                        ) : (
+                            <Typography className={classes.eulaContent} variant="subtitle1">
+                                {eulaDetails.loadingMessage}
+                            </Typography>
+                        )}
```

Comment:
You could simplify this:
```suggestion
                    <Typography className={classes.eulaContent} variant={eulaLoaded ? 'body1' : 'subtitle1'}>
                        {eulaLoaded ? eulaDetails.eulaContent : eulaDetails.loadingMessage}
                    </Typography>
```